pwndbg/aglib/kernel/__init__.py

```diff
@@ -130,36 +110,28 @@ def get_first_kernel_ro() -> pwndbg.lib.memory.Page | None:
     return None
 
 
-def load_kconfig() -> pwndbg.lib.kernel.kconfig.Kconfig | None:
+@pwndbg.lib.cache.cache_until("start")
+def kconfig() -> pwndbg.lib.kernel.kconfig.Kconfig | None:
+    global _kconfig
+    config_start, config_end = None, None
     if has_debug_syms():
         config_start = pwndbg.aglib.symbol.lookup_symbol_addr("kernel_config_data")
         config_end = pwndbg.aglib.symbol.lookup_symbol_addr("kernel_config_data_end")
     else:
         mapping = get_first_kernel_ro()
         results = list(pwndbg.search.search(b"IKCFG_ST", mappings=[mapping]))
 
-        if len(results) == 0:
-            return None
-
-        config_start = results[0] + len("IKCFG_ST")
-        config_end = list(pwndbg.search.search(b"IKCFG_ED", start=config_start))[0]
-
+        if len(results) != 0:
+            config_start = results[0] + len("IKCFG_ST")
+            config_end = list(pwndbg.search.search(b"IKCFG_ED", start=config_start))[0]
```

Comment:
Both this and `results` should probably do:

`next(pwndbg.search.search(...), None)` and instead of checking `len(..)` then check it as `is not None`

Doing it this way, we won't search all memory for those values as well as we won't render a Python list for the returned results.